costmap_2d/include/costmap_2d/layered_costmap.h

```diff
@@ -154,15 +163,63 @@ class LayeredCostmap
    * This is updated by setFootprint(). */
   double getInscribedRadius() { return inscribed_radius_; }
 
+protected:
+  /**
+   * ! Function copied from Costmap2D !
+   * To-Do: Modify function for this use-case
+   * @brief  Copy a region of a source map into a destination map
+   * @param  source_map The source map
+   * @param sm_lower_left_x The lower left x point of the source map to start the copy
+   * @param sm_lower_left_y The lower left y point of the source map to start the copy
+   * @param sm_size_x The x size of the source map
+   * @param  dest_map The destination map
+   * @param dm_lower_left_x The lower left x point of the destination map to start the copy
+   * @param dm_lower_left_y The lower left y point of the destination map to start the copy
+   * @param dm_size_x The x size of the destination map
+   * @param region_size_x The x size of the region to copy
+   * @param region_size_y The y size of the region to copy
+   */
+  template<typename data_type>
+    void copyMapRegion(data_type* source_map, unsigned int sm_lower_left_x, unsigned int sm_lower_left_y,
+                       unsigned int sm_size_x, data_type* dest_map, unsigned int dm_lower_left_x,
+                       unsigned int dm_lower_left_y, unsigned int dm_size_x, unsigned int region_size_x,
+                       unsigned int region_size_y)
+    {
+      // we'll first need to compute the starting points for each map
+      data_type* sm_index = source_map + (sm_lower_left_y * sm_size_x + sm_lower_left_x);
+      data_type* dm_index = dest_map + (dm_lower_left_y * dm_size_x + dm_lower_left_x);
+
+      // now, we'll copy the source map into the destination map
+      for (unsigned int i = 0; i < region_size_y; ++i)
+      {
+        memcpy(dm_index, sm_index, region_size_x * sizeof(data_type));
+        sm_index += sm_size_x;
+        dm_index += dm_size_x;
+      }
+    }
+
 private:
-  Costmap2D costmap_;
+
+  // Struct to store the bounds for each timed costmaps
+  struct Costmap2DBounds
+  {
+    double minx, miny, maxx, maxy;
+    int bx0, bxn, by0, byn;
```

Comment:
```suggestion
    unsigned int bx0, bxn, by0, byn;
```
F: keep the  same type as before

Reply:
only resolve conversations once having actually pushed the corresponding changes

Reply:
ah sorry about that